Toggl.Foundation.MvvmCross/ViewModels/SuggestionsViewModel.cs

```diff
@@ -16,92 +19,63 @@ namespace Toggl.Foundation.MvvmCross.ViewModels
+        public InputAction<Suggestion> StartTimeEntryAction { get; private set; }
 
-        public bool IsEmpty => Suggestions.None();
-
-        public MvxAsyncCommand<Suggestion> StartTimeEntryCommand { get; set; }
+        private readonly IInteractorFactory interactorFactory;
+        private readonly IOnboardingStorage onboardingStorage;
+        private readonly ISuggestionProviderContainer suggestionProviders;
+        private readonly ISchedulerProvider schedulerProvider;
 
         public SuggestionsViewModel(
             ITogglDataSource dataSource,
             IInteractorFactory interactorFactory,
             IOnboardingStorage onboardingStorage,
-            ISuggestionProviderContainer suggestionProviders)
+            ISuggestionProviderContainer suggestionProviders,
+            ISchedulerProvider schedulerProvider)
         {
             Ensure.Argument.IsNotNull(dataSource, nameof(dataSource));
             Ensure.Argument.IsNotNull(interactorFactory, nameof(interactorFactory));
             Ensure.Argument.IsNotNull(suggestionProviders, nameof(suggestionProviders));
             Ensure.Argument.IsNotNull(onboardingStorage, nameof(onboardingStorage));
+            Ensure.Argument.IsNotNull(schedulerProvider, nameof(schedulerProvider));
 
-            this.dataSource = dataSource;
             this.interactorFactory = interactorFactory;
             this.onboardingStorage = onboardingStorage;
             this.suggestionProviders = suggestionProviders;
-
-            StartTimeEntryCommand = new MvxAsyncCommand<Suggestion>(startTimeEntry, _ => areStartButtonsEnabled);
+            this.schedulerProvider = schedulerProvider;
         }
 
-        public async override Task Initialize()
+        public override async Task Initialize()
         {
             await base.Initialize();
 
-            emptyDatabaseDisposable = dataSource
-                .TimeEntries
-                .IsEmpty
-                .FirstAsync()
-                .Subscribe(fetchSuggestions);
-        }
-
-        private void fetchSuggestions(bool databaseIsEmpty)
-        {
-            Suggestions.Clear();
-
-            if (databaseIsEmpty) return;
+            StartTimeEntryAction = InputAction<Suggestion>.FromAsync(suggestion => startTimeEntry(suggestion));
 
-            suggestionProviders
+            Suggestions = suggestionProviders
                 .Providers
                 .Select(provider => provider.GetSuggestions())
                 .Aggregate(Observable.Merge)
-                .Subscribe(addSuggestion);
-        }
-
-        private void addSuggestion(Suggestion suggestion)
-        {
-            Suggestions.Add(suggestion);
-
-            RaisePropertyChanged();
+                .ToArray()
+                .AsDriver(onErrorJustReturn: new Suggestion[0], schedulerProvider: schedulerProvider);
+
+            IsEmpty = Suggestions
+                .Select(suggestions => suggestions.Length)
+                .Select(c => c == 0)
```

Comment:
I think this is better as just one `Select`
```suggestion
                .Select(suggestions => suggestions.Length == 0)
```